src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,129 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidInputDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "my_valid_login";
+    private static final String VALID_PASSWORD = "valid#$PASS@!1234";
+    private static final int MINIMUM_LENGTH = 6;
+    private static final int VALID_AGE_MINIMUM = 18;
+    private static final int AGE_ZERO_INVALID = 0;
+
+    private static RegistrationServiceImpl registrationService;
+    private User user;

```

Comment:
```suggestion
    private final RegistrationServiceImpl registrationService;
    private User validUser; // because we might have several variants of user in this test class in the future
```